champselect.py

```
@@ -62,50 +63,94 @@ def lock_champ(connection: c.Connection, champid: int | None = None) -> None:
+        if time.time() > start_time + connection.lock_in_delay:
+            still_waiting = False
+
+        # Make sure we update the hover if champ is changed by web API
+        update_champselect(connection)
+        should_rehover = action.update_champid()
+        if should_rehover:
+            with action:
+                _do_champ_inner(connection, action)
+
+        # Check if user manually completed the action
+        if (action.banning() and connection.ban_action["completed"]
```

Ruff (0.14.0)

Parenthesize `a and b` expressions when chaining `and` and `or` together, to make the precedence clear

Parenthesize the `and` subexpression

(RUF021)